ui/app/components/container-image-tag/index.ts

```diff
@@ -0,0 +1,64 @@
+import Component from '@glimmer/component';
+import { inject as service } from '@ember/service';
+import ApiService from 'waypoint/services/api';
+import { StatusReport } from 'waypoint-pb';
+
+interface Args {
+  statusReport: StatusReport.AsObject;
+}
+
+export default class DockerImageBadge extends Component<Args> {
+  @service api!: ApiService;
+
+  get states(): unknown {
+    return this.args.statusReport.resourcesList
+      ? this.args.statusReport.resourcesList.map((r) => JSON.parse(r.stateJson ?? '{}'))
+      : [];
+  }
+
+  get imageRefs(): ImageRef[] {
+    return findImageRefs(this.states);
+  }
+}
+
+function findImageRefs(obj: unknown, result: ImageRef[] = []): ImageRef[] {
+  if (typeof obj !== 'object') {
+    return result;
+  }
+
+  if (obj === null) {
+    return result;
+  }
+
+  for (let [key, value] of Object.entries(obj)) {
+    if (key.toLowerCase() === 'image' && typeof value === 'string') {
+      if (!result.some((image) => image.ref === value)) {
+        result.push(new ImageRef(value));
+      }
+    } else {
+      findImageRefs(value, result);
+    }
+  }
+
+  return result;
+}
+
+class ImageRef {
```

Comment:
For consistency with the structure of other files, could we move the `ImageRef` class definition above the Component class?